web/src/components/common/Modal.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { X } from 'lucide-react';
+
+interface ModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  title: string;
+  children: React.ReactNode;
+  footer?: React.ReactNode;
+}
+
+export const Modal: React.FC<ModalProps> = ({ isOpen, onClose, title, children, footer }) => {
+  if (!isOpen) return null;
```

Comment:
As it stands, won't this render a `null` in our DOM if `isOpen` is false?

I wonder if the decision to render this should be at the top level component instead, similar to things like:
- https://github.com/replicatedhq/embedded-cluster/blob/5c52173fa644edea3aa6dc4026063d5774d99e4e/web/src/components/wizard/InstallationStep.tsx#L94-L95
- https://github.com/replicatedhq/embedded-cluster/blob/29678df88daa873458a08e7394f0b434a1951ee9/web/src/components/wizard/installation/LogViewer.tsx#L39-L51